### ISSUE TITLE
Cant Seem to load as integration

### ISSUE THREAD

**Issue by broswell** [uhppoted/uhppoted-app-home-assistant#8]
I am running under docker.  I have the following files under by custom_components directory

root@hassio:/home/homeassistant/.homeassistant/custom_components# ls
hacs  __init.py__  sonoff  uhppoted  uhppoted-app-homeassistant  uhppoted-app-home-assistant-0.8.9  v0.8.9.tar

and I have a copy 
root@hassio:/home/homeassistant/.homeassistant/config/custom_components# ls -l
total 4
-rwxrwxrwx 1 root root  0 Nov 26 13:26 __init.py__
lrwxrwxrwx 1 root root 65 Nov 26 13:28 uhppoted -> /home/pi/uhppoted-app-home-assistant/custom_components//uhppoted/
lrwxrwxrwx 1 root root  8 Nov 26 13:58 uhppoted-app-homeassistant -> uhppoted

Cant figure out how to install the integration. Only option seems to be search by name, and I don't find anything beginning with uhp

Could anyone help get me started?
Thanks!
broswell@syssrc.com

**Comment by uhppoted** [comment]
Hi Bob,

@YBOnline successfully installed it as a HACS plugin by using a _custom repository_ - that might be the simplest thing to try first. The instructions are in one of the [discussions](https://github.com/uhppoted/uhppoted-app-home-assistant/discussions/4#discussioncomment-8934289).

**Comment by uhppoted** [comment]
If the HACS _custom repository_ doesn't work for you then this is what your directory structure should look like to use the dev version from _github_ (or the files extracted from the release _.tar.gz_ file):
```
hass
   homeassistant
   ...
   ...
   config
      custom_components
         __init__.py
         uhppoted
             __init__.py
            manifest.json
            services.yaml
            uhppoted.py
            button.py
            ...
            ...
            coordinators
               coordinators.py
               controllers.py
               ...
            services
               services.py
               ...
            translations
               en.json
               ...
```

**Comment by broswell** [comment]
In my setup, the files had to go into the directory
/home/homeassistant/.homeassistant/custom_components and not
/home/homeassistant/.homeassistant/config/custom_components

I got it to show up in homeassistant, and got it find my controller, but the joy ended there.
In the Integration screen it says "no devices or entities"  ! Not loaded.  

The notifications Log details error
Logger: homeassistant.setup
Source: setup.py:269
First occurred: 6:54:22 PM (1 occurrences)
Last logged: 6:54:22 PM

Setup failed for custom integration 'uhppoted': No setup or config entry setup function defined.
Not sure where to go from here.

Thanks

**Comment by uhppoted** [comment]
In your startup logs, do you get anything like this:
```
...
2024-11-26 17:00:43.476 INFO (MainThread) [homeassistant.setup] Setting up uhppoted
2024-11-26 17:00:43.476 INFO (MainThread) [custom_components.uhppoted] default bind address:        192.168.1.100
2024-11-26 17:00:43.476 INFO (MainThread) [custom_components.uhppoted] default broadcast address:   192.168.1.255:60000
2024-11-26 17:00:43.476 INFO (MainThread) [custom_components.uhppoted] default listen address:      192.168.1.100:60001
2024-11-26 17:00:43.476 INFO (MainThread) [custom_components.uhppoted] default debug:               False
2024-11-26 17:00:43.476 INFO (MainThread) [custom_components.uhppoted] default timezone:            CEST
2024-11-26 17:00:43.476 INFO (MainThread) [custom_components.uhppoted] default timeout:             1.23s
...
```
You might need to add an entry to your _configuration.yaml_ file (under the config folder in my setup), e.g.:
```
...
uhppoted:
    bind_address: 0.0.0.0
    broadcast_address: 255.255.255.255:60000
    listen_address: 0.0.0.0:60001
    debug: false
...
```

**Comment by broswell** [comment]
I had the configuration.yaml entries, and I took them out as I got an error saying something like "this integration does not support YAML"
I'll try again.  I don't get the entries you have.  I put the entries back in configuration.yaml.
Looking at the log


***@***.******@***.***:/home/homeassistant/.homeassistant#> grep uhpp *.log*
home-assistant.log:2024-11-27 01:13:59.453 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration uhppoted which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
home-assistant.log:2024-11-27 01:14:02.718 ERROR (MainThread) [homeassistant.setup] The 'uhppoted' integration does not support YAML setup, please remove it from your configuration
home-assistant.log:2024-11-27 01:14:02.719 ERROR (MainThread) [homeassistant.setup] Setup failed for custom integration 'uhppoted': No setup or config entry setup function defined.

**Comment by broswell** [comment]
*   Core2024.11.3
  *   Frontend  20241106.2
Running as a docker container on a PI 5
I'd love it if someone could remote in with me for a few minutes and show me what I am doing wrong.  In return, I'm happy to code additional functionality in Python.


Bob

**Comment by uhppoted** [comment]
Ok, so basically the latest and greatest. 

Not convinced you are doing anything wrong - it may be well be something to do with Docker and/or the Docker image because that's an an untested configuration. Let me spend some time to get a Docker image up and running and see if it replicates the problem and we can take it from there.

I'll keep you posted .. it might take a little while though (days not weeks :-)).

**Comment by broswell** [comment]
Thank you.  If you celebrate, heres to a great holiday!

**Comment by YBonline** [comment]
> Hi Bob,
> 
> @YBonline successfully installed it as a HACS plugin by using a _custom repository_ - that might be the simplest thing to try first. The instructions are in one of the [discussions](https://github.com/uhppoted/uhppoted-app-home-assistant/discussions/4#discussioncomment-8934289).

Just figured I'd pipe up with my current information since I was mentioned:
I currently have Installed version   ea7016d
It has some time ago alerted me to updates since then, and HACS/HA refuses to install the update at this point (trying to install 655e41e)

If memory serves me right, I originally installed with HACS 1.x, and I know HACS 2.0 was a major rewrite which I am running now. Not sure if the issue is related to HACS 2.0 or uhppoted-app-home-assistant... I have not had any time to look into this issue at all as I've been involved with Hurricane Helene/Milton cleanup efforts

**Comment by uhppoted** [comment]
Ah! Thanks @YBonline - I dug into it a bit and it turns out that HACS now requires a _hacs.json_ file (and no doubt a whole bunch of other stuff) which probably has something to do with the rewrite. I've created an [issue](https://github.com/uhppoted/uhppoted-app-home-assistant/issues/10) to look into it.

Good luck with the cleanup - hope you weren't too badly impacted (Helene in particular looked horrific).

**Comment by uhppoted** [comment]
> Thank you. If you celebrate, heres to a great holiday! 

Thanks @broswell .. Thanksgiving here was a month ago but it was nonetheless fun :-). Hope yours is good too!

Ok, onto more mundane things - I've managed to get _uhppoted-app-home-assistant_ up and running on a Docker container. It wasn't particularly happy until the network was set to _bridge_ mode - AFAIK that's probably because MacOS doesn't support _host_ mode, but you could give it a try.

For reference this is what my Docker compose file looks like:
```
services:
  home-assistant:
    container_name: home-assistant
    image: "ghcr.io/home-assistant/home-assistant:stable"
    volumes:
      - /config
      - ./uhppoted:/config/custom_components/uhppoted
    restart: unless-stopped
    privileged: false
    network_mode: "bridge"
    ports:
       - "8123:8123"
```

The bad news is that _Home Assistant_ has changed something in the way they do configuration and it's now spewing warnings and errors that definitely weren't there before and is more or less unusable. I'm going to have to look into it and see what exactly is happening &lt;sigh&gt;:
- https://github.com/uhppoted/uhppoted-app-home-assistant/issues/12
- https://github.com/uhppoted/uhppoted-app-home-assistant/issues/11

I'll post updates here as/when there is anything to try out.

**Comment by broswell** [comment]
Thank you for looking into this!  I'm setting up the access control system in our new warehouse.  For now, I'm able to use the system using the provided windows software and the uhpotte-cli interfaces which get the job done.  Looking forward to integrating into home assistant ASAP.

Bob

**Comment by uhppoted** [comment]
Warehouse? Interesting ... &lt;smile&gt; clearly nobody takes the _Home_ in _Home Assistant_ very seriously!

**Comment by broswell** [comment]
I am using Home Assistant in a 72,000 sq foot (6,600 sq meter)commercial building.  HA controls/monitors  thermostats for 14  of the roof top units, turns on and off about 20 smart switches for displays in our museum, monitors temperature and humidity in high value artifacts, (Apple 1, Cray-1, Kenbek-1 etc) provides a secondary alarm system, and monitors leak detectors throughout the building.   It  also tracks the output of our solar plant and grid electric usage.

If you are in the Baltimore, Maryland area, visit us at the Computer Museum @ System Source (https://museum.syssrc.com)

**Comment by uhppoted** [comment]
Hi,

I've just released [v0.8.9.3](https://github.com/uhppoted/uhppoted-app-home-assistant/releases/tag/v0.8.9.3) which:
- Fixes the deprecation warnings in _Home Assistant_ 2024.11.3
- Adds the _hacs.json_ file for HACS v2.0
- Includes sample _Docker_ `compose.yml` files.

I've tested installation from HACS and it seems to work fine and the functionality all seems to work - I did notice that _Home Assistant_ needed to be restarted after the initial _uhppoted_ configuration before it would register some of the _sensor_ entities. Anyway it's there if you'd like to try it out and let me know how you go..

I feel I should remind you that it's still very much an _alpha_ release - functional but a very long way from either finished or pretty. In particular the new _Home Assistant_ feature which automatically populates the _Overview_ page is a bit disconcerting - it takes a fair bit of manual editing to get a reasonably informative page.
